### DISCUSSION TITLE
I can't seem to get Chat GPT to have access to my notes

### DISCUSSION THREAD

**Discussion by Krysem** [brianpetro/obsidian-smart-connections#128] [Q&A]
As far as I can tell I've gotten the embeddings and smart connections pane to work as intended. But when it comes to Smart Chat, it doesn't seem to be able to answer any questions about the notes in my vault. I only get generic responses equivalent to asking the same prompts via Chat-GPTs web interface. 

Is there something obvious that I am missing in order to have the chat be able to search, analyze and understand the context of my notes?

**Comment by brianpetro** [comment]
Have you tried starting your questions with "Based on my notes..."?

**Comment by Krysem** [comment]
That definitely worked for accessing the notes! I guess I just need to include something along the lines of referencing the notes for any prompt given. Thank you!

**Comment by brianpetro** [comment]
@Krysem great to hear that it worked!

And yes, the "context lookup" is currently triggered by pronouns. You can see which ones in the language settings for Smart Connections. 

Sorry for the confusion!

Brian 🌴

**Reply by GBSpada** [reply]
I find that smart chat is too slow and I don't think that my vault is big enough for it to slow down
![Uploading Screenshot 2023-03-31 125728.png…]()

**Reply by brianpetro** [reply]
@GBSpada if you're aware of the specific note that you'd like to use as context, you can specify the note using the bracket [[link]] syntax. This is much faster than the process used when it searches your entire vault, which takes a similar amount of time regardless of your vault size because it relies on additional completions from the API.

**Reply by brianpetro** [reply]
Note: the specifying notes using their link is new in version `1.5.0`.

**Reply by GBSpada** [reply]
Ok now it works but it's kind of limiting considering that it does not search inside linked files. Great plulgin tho

**Reply by brianpetro** [reply]
@GBSpada what do you mean by this?

> does not search inside linked files

Do you mean notes that are linked from the note that you specified?

Soon I'll be implementing something like "specify by folder" as context.

There are also some additional ways I'd like to improve the ability to gather all the accurate context. However, these will increase the number of processes needing to be completed which would further slow down the time until you get results.

Maybe logging the completed processes, like in ChatGPT plugins, would be sufficient since it's be letting you know what's happening in the background. What do you think?

![Screenshot_20230401-113302](https://user-images.githubusercontent.com/1886014/229298750-45ef4c12-dd37-44ac-843d-bc6e916e1b97.png)

Thanks for following up,
Brian 🌴

**Reply by GBSpada** [reply]
Yes, i mean notes that are linked from the note that i have specified. Looking forward to these new implementations, you're doing god's work with this plugin

**Reply by brianpetro** [reply]
Thanks for following up and the kind words!

I will keep that in mind for future updates.

Brian 🌴
